### ISSUE TITLE
Make View() a "background" task so it can still be run when the console is busy

### ISSUE THREAD

**Issue by francisbarton** [rstudio/rstudio#9668]
### Use case:
I am waiting for a long-running command to complete, and let's say I want to use this time to have a look at an object from my environment, eg inspect a data frame.

### Problem:
Currently if I click on the object in the Environment pane, this calls `View()` – but this uses the console, and can't run until the console is free again, so I have to wait.
Yet, by comparison, if I click on a file in the Files pane, this also opens a new tab in the Source pane, but without needing to wait for the console.

I understand that opening a file and `View()`ing an object are not the same, but I think it would be an improvement if RStudio could still run `View(x)` "in the background" without needing to wait for the Console.

**Comment by ronblum** [comment]
@francisbarton Thank you for the suggestion! I'll mark it for review as part of our ongoing development of RStudio.

**Comment by jmcphers** [comment]
This isn't really possible under the current architecture because `View()` uses R itself to format, filter, and process the values. Because R is not multi-threaded and doesn't handle concurrency, that all has to happen on the main thread. (And shipping the data down to a sub-process would be more just as expensive or more; keep in mind that these frames are often multi-GB in size.)

**Comment by francisbarton** [comment]
Ah well, that is a shame. It was a "job" type approach that I was thinking about. But I didn't know about the internal limitations. Feel free to close.

**Comment by jmcphers** [comment]
No, we'll keep this open, just wanted to signal that it's more work than it looks like (will require some changes in architecture).
